### PR TITLE
RIA-3218 Legal rep apply for a late remission

### PR DESCRIPTION
**Pull request by tarun-palisetty**
### JIRA link (if applicable) ###
[RIA-3218](https://tools.hmcts.net/jira/browse/RIA-3218)

### Change description ###
Legal rep apply for a late remission

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RequestFeeRemissionConfirmation.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        PostSubmitCallbackResponse postSubmitResponse =
+            new PostSubmitCallbackResponse();
+
+        postSubmitResponse.setConfirmationHeader("# Your remission request has been submitted");
+        postSubmitResponse.setConfirmationBody(
+            "#### What happens next\n\n"
+            + "The Tribunal will review the remission request. The decision details will be available to view in the appeal tab "
+            + "when the decision has bee made.<br>"
```
Comment (divisathyan) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RequestFeeRemissionConfirmation.java:37]: There is a typo. Should it be "has been made"

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RequestFeeRemissionConfirmationTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.postsubmit;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.assertj.core.api.Assertions.*;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertFalse;
```
Comment (divisathyan) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RequestFeeRemissionConfirmationTest.java:6]: Remove junit4 assert

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RequestFeeRemissionConfirmationTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        assertNotNull(callbackResponse);
+        assertTrue(callbackResponse.getConfirmationHeader().isPresent());
+        assertTrue(callbackResponse.getConfirmationBody().isPresent());
+
+        assertThat(callbackResponse.getConfirmationHeader().get())
+            .contains("# Your remission request has been submitted");
+        assertThat(callbackResponse.getConfirmationBody().get())
+            .contains("#### What happens next\n\n"
+                      + "The Tribunal will review the remission request. The decision details will be available to view in the appeal tab "
+                      + "when the decision has bee made.<br>");
```
Comment (divisathyan) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RequestFeeRemissionConfirmationTest.java:52]: Typo correction above should reflect here

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestFeeRemissionHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
```
Comment (divisathyan) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestFeeRemissionHandlerTest.java:5]: Remove junit4 imports

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestFeeRemissionPreparerTest.java [hunk 5]
```diff
@@ -0,0 +1,848 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.Assert.assertFalse;
```
Comment (divisathyan) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RequestFeeRemissionPreparerTest.java:4]: Remove junit4 imports

**Review by divisathyan — CHANGES_REQUESTED** [review]
Please see comments

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RequestFeeRemissionConfirmationTest.java (hunk 3 above)
Comment (tarun-palisetty) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RequestFeeRemissionConfirmationTest.java]: removed

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Review by divisathyan — APPROVED** [review]

**Pull request merged**